### PR TITLE
Feature/sc 332/preview applicant image when uploading

### PR DESCRIPTION
**Pull request by Plebbimon**
relates to https://github.com/KSG-IT/ksg-nett/tree/feature/sc-438/admission-misc-fixes

### CODE REVIEW

File: src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx [hunk 1]
```diff
@@ -58,12 +76,19 @@ export const SetPriorities: React.VFC<SetPrioritiesProps> = ({
...
+  }, [internalGroupPositions, applicant.priorities])
 
+  useEffect(() => {
+    const filteredPriorities = values.filter(priority => priority !== null)
+    const priorityIds = filteredPriorities.map(
+      priority => priority!.internalGroupPosition.id
+    )
+    handleUpdatePriorities(priorityIds)
+    setIsDirty(false)
+  }, [isDirty])
```
Comment (alexaor) [inline src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx:91]: Does this need a useEffect? Internal state in most cases are manageable with eventhandlers

File: src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx [hunk 2]
```diff
@@ -78,30 +103,80 @@ export const SetPriorities: React.VFC<SetPrioritiesProps> = ({
...
+      },
+    })
+  }
+
+  const handlePriorityChange = (index: number, direction: -1 | 1) => {
+    handlers.reorder({ from: index, to: index + direction })
+    setIsDirty(true)
+  }
+  const handleUpdatePriorities = (priorityIds: string[]) => {
+    console.table(priorityIds)
```
Comment (alexaor) [inline src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx:141]: remove console log

File: src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx [hunk 3]
```diff
@@ -78,30 +103,80 @@ export const SetPriorities: React.VFC<SetPrioritiesProps> = ({
...
+    updatePriorities({
+      variables: {
+        priorityOrder: priorityIds,
+        applicantId: applicant.id,
+        token: applicantToken,
+      },
+      onError(message) {
+        showNotification({
+          title: 'Error',
+          message: `${message}`,
```
Comment (alexaor) [inline src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx:151]:
```suggestion
      onError({message}) {
        showNotification({
          title: 'Error',
          message
```

File: src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx [hunk 4]
```diff
@@ -78,30 +103,80 @@ export const SetPriorities: React.VFC<SetPrioritiesProps> = ({
...
+        applicantId: applicant.id,
+        token: applicantToken,
+      },
+      onCompleted() {
+        showNotification({
+          title: 'Priority deleted',
+          message: 'Your priority has been deleted',
+          color: 'teal',
+        })
+      },
```
Comment (alexaor) [inline src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx:171]: Remove onCompleted

File: src/modules/admissions/components/ApplicantPortal/components/InterviewsAvailableForBooking.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  Title,
+  Text,
+  SimpleGrid,
+  Divider,
+} from '@mantine/core'
+import { format } from '../../../../../util/date-fns'
+import {
+  AvailableInterviewsDayGrouping,
+  InterviewsAvailableForBookingReturns,
+} from '../../../types.graphql'
```
Comment (alexaor) [inline src/modules/admissions/components/ApplicantPortal/components/InterviewsAvailableForBooking.tsx:15]: Use absolute paths

Comment (alexaor) [inline src/modules/admissions/components/ApplicantPortal/components/InterviewsAvailableForBooking.tsx:15]: Use absolute paths

File: src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx [hunk 6]
```diff
@@ -78,30 +103,80 @@ export const SetPriorities: React.VFC<SetPrioritiesProps> = ({
...
     AddInternalGroupPositionPriorityVariables
-  >(ADD_INTERNAL_GROUP_POSITION_PRIORITY, {
+  >(APPLICANT_ADD_INTERNAL_GROUP_POSITION_PRIORITY, {
     refetchQueries: ['GetApplicantFromToken'],
   })
 
+  const [updatePriorities] = useMutation<
+    UpdateInternalGroupPositionPriorityOrderReturns,
+    ApplicantUpdateInternalGroupPositionPriorityOrderVariables
+  >(APPLICANT_UPDATE_INTERNAL_GROUP_POSITION_PRIORITY_ORDER_MUTATION)
```
Comment (alexaor) [inline src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx:123]: Move thee mutations to a hook

**Review by alexaor — CHANGES_REQUESTED** [review]

**Comment by alexaor** [comment]
Is there a backend PR missing here as well?

**Comment by Plebbimon** [comment]
https://github.com/KSG-IT/ksg-nett/tree/feature/sc-438/admission-misc-fixes

File: src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx (hunk 1 above)
Comment (Plebbimon) [reply src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx:91]: I don't know. Couldn't make it work as I wanted it to when putting all the logic in a handler that both reorders and updates, possibly due to how `use-list-state`'s handler reorders elements.

File: src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx (hunk 2 above)
Comment (Plebbimon) [reply src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx:141]: oopsie

File: src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx (hunk 3 above)
Comment (Plebbimon) [reply src/modules/admissions/components/ApplicantPortal/SetPriorities.tsx:151]: I'm guessing that we don't actually want to display the errors to the user via a notification.

**Review by alexaor — APPROVED** [review]

**Pull request merged**